numpy/f2py/tests/test_quoted_character.py

```diff
@@ -28,3 +33,97 @@ class TestQuotedCharacter(util.F2PyTest):
                         reason='Fails with MinGW64 Gfortran (Issue #9673)')
     def test_quoted_character(self):
         assert_equal(self.module.foo(), (b"'", b'"', b';', b'!', b'(', b')'))
+
+@pytest.mark.parametrize("extra_args", [
+    # extra_args can be a list as of gh-11937
+    (['--noopt', '--debug']),
+    # test for string as well, using the same
+    # fcompiler options
+    ('--noopt --debug'),
+    ])
+def test_f2py_init_compile(extra_args):
+    # flush through the f2py __init__
+    # compile() function code path
+    # as a crude test for input handling
+    # following migration from exec_command()
+    # to subprocess.check_output() in gh-11937
+
+    fsource =  '''
+             integer function foo()
+             foo = 10 + 5
+             return
+             end
+             '''
+    # use various helper functions in util.py to
+    # enable robust build / compile and
+    # reimport cycle in test suite
+    d = util.get_module_dir()
+    modulename = util.get_temp_module_name()
+
+    # mimic the path changing behavior used
+    # by build_module() in util.py, but don't
+    # actually use build_module() because it
+    # has its own invocation of subprocess
+    # that circumvents the f2py.compile code
+    # block under test
+    cwd = os.getcwd()
+    target = os.path.join(d, str(uuid.uuid4()) + '.f')
+    with open(target, 'w') as target_file:
+
+        try:
+            os.chdir(d)
+            ret_val = numpy.f2py.compile(fsource,
+                                         modulename=modulename,
+                                         extra_args=extra_args,
+                                         source_fn=target)
+        finally:
+            os.chdir(cwd)
+
+        # check for compile success return value
+        # NOTE: this is as far as we currently
+        # expect Windows-based test to proceed,
+        # with expected return value of 1 because
+        # of lack of Fortra compiler; it is better 
+        # to require control flow to reach this point
+        # than to completely skip the test on windows
+        try:
+            assert_equal(ret_val, 0)
+        except AssertionError:
+            if os.name == 'nt' and sys.version_info[0] < 3:
+                pytest.skip("May fail on windows with Python 2.x")
+
+        if os.name != 'nt':
```

Comment:
Should this condition also by python 2.x? If so, I'd move this inside the `try` above.